addons/mail/static/src/model/model_manager.js

```diff
@@ -260,6 +262,29 @@ export class ModelManager {
         return res;
     }
 
+    /**
+     * Returns the messaging singleton associated to this model manager.
+     *
+     * @returns {mail.messaging}
+     */
+    get messaging() {
+        if (!this.models || !this._messaging) {
+            return undefined;
+        }
+        // Use "get" specifically to ensure the record still exists and to
+        // ensure listeners are properly notified of this access.
+        return this.get(this.models['mail.messaging'], this._messaging.localId);
+    }
+
+    /**
+     * Returns all generated models.
+     *
+     * @returns {Object[]} keys are model name, values are model class
+     */
+    get models() {
+        return this._models;
+    }
```

Comment:
I would prefer no getter and just return the attribute (and remove the _ prefix).
Looks straight-forward for `models`.
For `messaging`, the tracking of listening access is essential so I wouldn't mind keeping the getter approach for the time being.

Reply:
ok